src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4851,13 +4851,47 @@ public PreprocessingSymbolInfo GetPreprocessingSymbolInfo(IdentifierNameSyntax n
 
             if (node.Ancestors().Any(n => SyntaxFacts.IsPreprocessorDirective(n.Kind())))
             {
-                bool isDefined = this.SyntaxTree.IsPreprocessorSymbolDefined(node.Identifier.ValueText, node.Identifier.SpanStart);
-                return new PreprocessingSymbolInfo(new Symbols.PublicModel.PreprocessingSymbol(node.Identifier.ValueText), isDefined);
+                return CreatePreprocessingSymbolInfo(node.Identifier);
             }
 
             return PreprocessingSymbolInfo.None;
         }
 
+        /// <summary>
+        /// Gets the preprocessing symbol info for the preprocessing symbol defined in the #define directive.
+        /// </summary>
+        /// <param name="node">A #define directive trivia node.</param>
+        private PreprocessingSymbolInfo GetPreprocessingSymbolInfo(DefineDirectiveTriviaSyntax node)
+        {
+            CheckSyntaxNode(node);
+            return CreatePreprocessingSymbolInfo(node.Name);
+        }
+
+        /// <summary>
+        /// Gets the preprocessing symbol info for the preprocessing symbol undefined in the #undef directive.
+        /// </summary>
+        /// <param name="node">An #undef directive trivia node.</param>
+        private PreprocessingSymbolInfo GetPreprocessingSymbolInfo(UndefDirectiveTriviaSyntax node)
+        {
+            CheckSyntaxNode(node);
+            return CreatePreprocessingSymbolInfo(node.Name);
+        }
```

Comment:
these methods seem unnecessary.  can you inline to caller (unless they're actually used in multiple places).

Reply:
or make local functions in GetPreprocessingSymbolInfoCore.

Reply:
Those two are used in the switch in `GetPreprocessingSymbolInfoCore`, L5103. What I find more useful is a method that does the simple check and create, since both syntax nodes share that same logic.